python/ray/autoscaler/_private/kubernetes/config.py

```diff
@@ -71,8 +71,11 @@ def fillout_resources_kubernetes(config):
         return config
     node_types = copy.deepcopy(config["available_node_types"])
     for node_type in node_types:
-        container_data = node_types[node_type]["node_config"]["spec"][
-            "containers"][0]
+
+        node_config = node_types[node_type]["node_config"]
+        pod = node_config.get("pod", node_config)
+        container_data = pod["spec"]["containers"][0]
+
```

Comment:
huh?

Reply:
Added a comment explaining what it's for.
Really, it's to prevent `kubernetes/example-ingress.yaml` failing `test_autoscaler_yaml.py`.